VerilogCodingStyle.md

```diff
@@ -1808,6 +1808,55 @@ reduction are significant enough to outweigh the possible synthesis mismatch
 issues that the use of `X` literals may entail (especially with the gate-counts
 available in today's technologies).
 
+
+#### Catching errors where invalid values are consumed
+
+For an internally-generated signal that could be invalid (but not driven to `X`)
+and is used to trigger some action (such as a register write-enable), it is
+recommened to add an assert to check that when the enable is true, the signal is
+valid. This triggers a simple to diagnose failure when an invalid value has been
+accidentally used.
+
+```systemverilog
+
```

Comment:
I can see what this first example is driving-at having read the X-related issues, but I had to read it a couple of times to get there.

In a way this boils down to the recommendation "write assertions to check that the design isn't wrong" (and the example here is an assertion checking that an AND gate functions as an AND gate).

I would maybe just leave it out, but I appreciate that it then isn't addressing all the issues in #13 .

Otherwise LGTM.

Reply:
Yeah this particular example is a bit basic, but I do think such obviously true assertions do have value as when someone fiddles with the logic that makes it obviously true and messes it up you have a quick failure rather than a bug that may lurk for a while longer.

Maybe I'm just worrying about problems that don't really occur in practice!

Reply:
Indeed, it spells out more specific guidance based the general recommendation above. But given the specific nature of the problem where X could be consumed, it is worth noting, hence LGTM as well.